### PR TITLE
resolving issue with network deployment

### LINKED ISSUE #19: New-CmAzIaasNetworking - deployment resource name is too long and errors out
Maximum string limit is 64. Change the deployment resource name string.

### PR DESCRIPTION
**Pull request by deepend-dev**
Closes #19 

It also resolves a sub-issue where subnet has nsg and routetable both but template implements only one of them.

### CODE REVIEW

File: Cloudmarque.Azure/Public/IaaS/networking/New-CmAzIaasNetworking.json [hunk 1]
```diff
@@ -62,9 +62,9 @@
                 ]
             }
         },
-        {   "condition": "[not(equals(parameters('SubnetArmObject')[copyIndex('nsgdeploycopy')].networkSecurityGroup, 'none'))]",
+        {   "condition": "[and(not(equals(parameters('SubnetArmObject')[copyIndex('nsgdeploycopy')].networkSecurityGroup, 'none')), equals(parameters('SubnetArmObject')[copyIndex('nsgdeploycopy')].routeTable, 'none'))]",
             "apiVersion": "2019-10-01",
-            "name": "[concat(copyIndex('nsgdeploycopy'), '-Add-nsg-', parameters('SubnetArmObject')[copyIndex('nsgdeploycopy')].networkSecurityGroup ,'-to-subnet-', parameters('SubnetArmObject')[copyIndex('nsgdeploycopy')].subnetName)]",
+            "name": "[concat('Add-nsg-','to-subnet-', copyIndex('nsgdeploycopy'))]",
```
Comment (tm-luke) [inline Cloudmarque.Azure/Public/IaaS/networking/New-CmAzIaasNetworking.json:67]: Add-nsg and to-subnet can be one string.

File: Cloudmarque.Azure/Public/IaaS/networking/New-CmAzIaasNetworking.json [hunk 2]
```diff
@@ -97,13 +97,13 @@
                 }
             }
         },
-        {   "condition": "[not(equals(parameters('SubnetArmObject')[copyIndex('routeTableDeploycopy')].routeTable, 'none'))]",
+        {   "condition": "[and(not(equals(parameters('SubnetArmObject')[copyIndex('routeTableDeploycopy')].routeTable, 'none')), equals(parameters('SubnetArmObject')[copyIndex('routeTableDeploycopy')].networkSecurityGroup, 'none'))]",
             "apiVersion": "2019-10-01",
-            "name": "[concat(copyIndex('routeTableDeploycopy'), '-Add-routeTable-', parameters('SubnetArmObject')[copyIndex('routeTableDeploycopy')].routeTable ,'-to-subnet-', parameters('SubnetArmObject')[copyIndex('routeTableDeploycopy')].subnetName)]",
+            "name": "[concat('Add-routeTable-','to-subnet-', copyIndex('routeTableDeploycopy'))]",
             "type": "Microsoft.Resources/deployments",
```
Comment (tm-luke) [inline Cloudmarque.Azure/Public/IaaS/networking/New-CmAzIaasNetworking.json:103]: Same here too for Add-routeTable-

Comment (deepend-dev) [reply Cloudmarque.Azure/Public/IaaS/networking/New-CmAzIaasNetworking.json:103]: Done

File: Cloudmarque.Azure/Public/IaaS/networking/New-CmAzIaasNetworking.json (hunk 1 above)
Comment (deepend-dev) [reply Cloudmarque.Azure/Public/IaaS/networking/New-CmAzIaasNetworking.json]: Done

**Comment by deepend-dev** [comment]
changes done

**Pull request merged**